api/api.ts

```diff
@@ -14,49 +14,83 @@ dotenvConfig()
 const authUrl = process.env.URL_AUTH_SERVICE;
 
 app.post('/upload/:tenant', async (req: express.Request, res: express.Response) => {
-    try {
-        if (req.headers['content-type'] !== 'application/octet-stream') {
-            throw new Error('Bad Request')
-        }
+    if (req.headers['content-type'] !== 'application/octet-stream') {
+        console.error(`Bad Request at ${new Date()}`)
+        res.status(400).send('Bad Request')
+        return
+    }
 
-        const authHeader = req.headers['authorization']
-        if (!authHeader) {
-            throw new Error('Missing Authorization header')
-        }
+    const authHeader = req.headers['authorization']
+    if (!authHeader) {
+        console.error(`Missing Authorization header at ${new Date()}`)
+        res.status(401).send('Missing Authorization header')
+        return
+    }
 
-        const token = authHeader?.split(' ')[1]
-        if (!token) {
-            throw new Error('Missing bearer token')
-        }
+    const token = authHeader?.split(' ')[1]
+    if (!token) {
+        console.error(`Request missing bearer token at ${new Date()}`)
+        res.status(401).send('Missing bearer token')
+        return
+    }
 
-        const authParams = new URLSearchParams({
-            token: token,
-            tenant: req.params.tenant
-        });
+    const authParams = new URLSearchParams({
+        token: token,
+        tenant: req.params.tenant
+    });
 
-        const response = await fetch(`${authUrl}?${authParams}`)
-        if (response.status !== 200) {
-            throw new Error('Error fetching Authentication API')
-        }
+    const authResponse = await makeAuth(authUrl, authParams)
+
+    if ('errorMessage' in authResponse) {
+        console.error(`Error authenticating user. Caused by:\n${authResponse.errorMessage} \nat ${new Date()}`)
+        res.status(authResponse.status).send('Error trying authentication.')
+        return
+    }
 
-        const filename: string = getFilename(req)
-        const tenantConfig: TenantConfig = await getTenantConfig(req.params.tenant)
+    if (didAuthFailed(authResponse)) {
```

Comment:
I guess this `if` is unnecesary. Can you show me one simple case where `errorMessage in authResponse` is `false` and `didAuthFailed` is true ?

Reply:
I got this now. The case where user has a expired token. But I still guess that it could be made in one single `if` . What d'U think of the approach below ? Removing completely didAuthFailed.
```js
    if (authResponse.status !== 200) {
        console.error(`Error authenticating user. Caused by:\n${ 'errorMessage' in authResponse ? authResponse.errorMessage: null} \nat ${new Date()}`)
        res.status(authResponse.status).send('Error trying authentication.')
        return
    }

```

Reply:
I agree.